extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -70,21 +83,30 @@ export async function getMarkdownLink(document: vscode.TextDocument, ranges: rea
 	const edits: vscode.SnippetTextEdit[] = [];
 	let placeHolderValue: number = ranges.length;
 	let label: string = '';
-	let smartPaste: boolean = false;
+	let smartPaste = { paste: false, title: false };
+
 	for (let i = 0; i < ranges.length; i++) {
+
+		let title = document.getText(ranges[i]);
+		//  const skinny_document: SkinnyTextDocument = {
```

Comment:
Right now you may not need `SkinnyTextDocument`. If you do need it in the future, the goal should be that `SkinnyTextDocument` is more specific version of `vscode.TextDocument` with only the essential properties. That means you should be able to pass a normal `vscode.TextDocument` into any place that takes a `SkinnyTextDocument`